util/datastructures.py

```diff
@@ -21,6 +23,15 @@ def as_list(self) -> List[str]:
         representation[1::2] = self._edges
         return representation
 
+    @staticmethod
+    def from_list(meta_path: List):
+        assert len(meta_path) % 2 != 0 or len(meta_path) == 0
```

Comment:
But shouldn't the list have an odd length? If we have `n` nodes then we need to have `n - 1` edges and we therefore have `2n - 1` elements which is always odd.

Reply:
Well that is exactly what it says :) 
the meta-path must not be divisible by 2 i.e. musnt be even